### DISCUSSION TITLE
Not rendering at all - Error 500

### DISCUSSION THREAD

**Discussion by ciscam** [DirectoryLister/DirectoryLister#697] [Help & Support] [answered]
Hi!

**Environment:**
I'm running an apache2-server with php7.4. There are three folders in the WebRoot:
* DirectoryLister (a fresh clone of this repo, modified only with .env-file)
* phpinfo (a dir with <?php phpinfo(); ?> in its index.php)
* content (the directory to be served when DirectoryLister works)

According to phpinfo(); on the same vServer, php-zip, php-xml (DOM) and php-fileinfo are enabled:
```
Zip | enabled
Zip version | 1.15.6
Libzip headers version | 1.5.1
Libzip library version | 1.5.1
```

```
XML Support | active
XML Namespace Support | active
libxml2 Version | 2.9.10
```

```
fileinfo support | enabled
libmagic | 537
```

**Results:**
All I get is a blank page when opening the DirectoryLister index.php. No loading time or latency. Apache-logs report the correct request and an error 500 response.

**Already tried:**
* Enabling app_debug and disabling display_readmes, zip_downloads in .env file
* all files are www-data:www-data owned
* cache-dir is empty
* changing php memory_limit from 128M to 512M
* no .htaccess file exists

For reference, beside the phpinfo()-file, Nextcloud works fine on another vServer of the same machine.

Any ideas?

**Comment by PHLAK** [comment]
Are you getting any errors you can share, maybe in your PHP error log? If so that would help us understand what's going on to be able to fix it.

However, if there are no errors I suspect there is an issue with your web server configuration. You mention that you're using Apache. Is mod_php installed and enabled?

**Comment by ciscam** [comment]
Thank you for the hint.

in /var/log/apache2/error.log I get the following two error messages on executing the /index.php:
```
[Wed Sep 15 18:36:14.765934 2021] [php7:warn] [pid 391930] [client $IP:55458] PHP Warning:  require(/var/www/webroot/DirectoryLister/app/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/webroot/DirectoryLister/index.php on line 7, referer: https://url.com/
[Wed Sep 15 18:36:14.766047 2021] [php7:error] [pid 391930] [client $IP:55458] PHP Fatal error:  require(): Failed opening required '/var/www/webroot/DirectoryLister/app/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/www/webroot/DirectoryLister/index.php on line 7, referer: https://url.com/
```
and indeed there is no /app/vendor directory in this git, but a require directive on line 7 in /index.php

After commenting out line 7 in /index.php, executing it yields this error:
```
[Wed Sep 15 18:45:25.381888 2021] [php7:error] [pid 392104] [client $IP:55614] PHP Fatal error:  Uncaught Error: Class 'Dotenv\\Dotenv' not found in /var/www/webroot/DirectoryLister/index.php:13\nStack trace:\n#0 {main}\n  thrown in /var/www/webroot/DirectoryLister/index.php on line 13, referer: https://url.com/
```

As for mod_php, the ubuntu package 'libapache2-mod-php7.4' is installed. There are symlinks 'php7.4.conf' and 'php7.4.load' (containing the directive `LoadModule php7_module /usr/lib/apache2/modules/libphp7.4.so`) in apache2/modules-enabled.

**Comment by ciscam** [comment, marked as answer]
I used a release now, instead of a git-clone, which contains the vendor-directory and it works.

Sorry for the bother, thank you for the product and active support :)

**Reply by PHLAK** [reply]
Glad you figured that out.
